segmenttree/segmenttreeimpl.go

```diff
@@ -118,7 +118,7 @@ func (tree *SegmentTreeImpl) insert(node *Node, tupleToInsert ValueIntervalTuple
 				if node.size()+1 > tree.branchingFactor {
 					node = node.split()
 				}
-				//intervals = node.getIntervals() // recalculate as they might have changed // TODO is this needed?
+				intervals = node.getIntervals() // recalculate as they might have changed
```

Comment:
Yes, it is needed. We check for the new length on [l124](https://github.com/meggima/dmds2022-segmenttree/pull/8/files#diff-9be3a9e2c627a956051fb72e668b9268fd5fd0350abc9d0736da12f6dbe009a9L124) and assign it on [l103](https://github.com/meggima/dmds2022-segmenttree/pull/8/files#diff-9be3a9e2c627a956051fb72e668b9268fd5fd0350abc9d0736da12f6dbe009a9L103)